File: algo_go/find-x.go
package main

import (
    "fmt"
    "io"
)

const MAX_LEN = 200;

func main() {
    var arr_len int;
    var arr [MAX_LEN] int;
    for {
        _, err := fmt.Scanf("%d", &arr_len);
        if err == io.EOF {
            break;
        }
        for i := 0; i < arr_len; i++ {
            fmt.Scanf("%d", &arr[i]);
        }

        var targ, j int;
        fmt.Scanf("%d", &targ);
        for j := 0; j < arr_len; j++ {
            if arr[j] == targ {
                fmt.Printf("%d\n", j);
                break;
            }
        }

        if j == arr_len {
            fmt.Printf("-1\n");
        }
    }
}

File: algo-c/oldfiles/find-x.c
#include <stdio.h>

#define MAX_LEN 210

int arr[MAX_LEN];

int main() {
    int n, x;
    while (scanf("%d", &n) != EOF) {
        for (int i = 0; i < n; i++) {
            scanf("%d", &arr[i]);
        }
    
    scanf("%d", &x);
    int k;
    for (k = 0; k < n; k++) {
        if (arr[k] == x) {
            printf("%d\n", k);
            break;
        }
    }

    if (k == n) printf("-1\n");
    }
    
    return 0;
}
